### PR TITLE
feat: #56 added 404 page

### PR DESCRIPTION
**Pull request by chelovek-zeml**

### CODE REVIEW

File: mw-webapp/src/pages/Page404/Page404.tsx
Comment (Howl404) [inline mw-webapp/src/pages/Page404/Page404.tsx:3]: In this case, where string is only used in one place and is unlikely to be reused, i don't see a reason to make it a variable, little complexity without benefit

File: mw-webapp/src/styles/_variables.scss
Comment (Howl404) [inline mw-webapp/src/styles/_variables.scss:17]: Start comment from uppercase

File: mw-webapp/src/pages/Page404/Page404.module.scss
Comment (Ekaterina1994) [inline mw-webapp/src/pages/Page404/Page404.module.scss:6]: why height of page's container is 70vh?

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/pages/Page404/Page404.tsx
Comment (Howl404) [reply mw-webapp/src/pages/Page404/Page404.tsx:3]: I think using a variable here is breaking KISS.

1. String appears only once
2. String will never change
3. Meaning of the string is obvious from the context

By this logic we should make every simple html text into variables

File: mw-webapp/src/styles/_variables.scss
Comment (chelovek-zeml) [reply mw-webapp/src/styles/_variables.scss:17]: Fixed.

File: mw-webapp/src/styles/_variables.scss
Comment (chelovek-zeml) [reply mw-webapp/src/styles/_variables.scss:22]: s - means size(font-size).

File: mw-webapp/src/pages/Page404/Page404.tsx
Comment (chelovek-zeml) [reply mw-webapp/src/pages/Page404/Page404.tsx:5]: You can't start a function name with a number.

File: mw-webapp/src/pages/Page404/Page404.module.scss
Comment (chelovek-zeml) [reply mw-webapp/src/pages/Page404/Page404.module.scss:6]: If you take the entire height of the screen, scrolling appears.

File: mw-webapp/src/pages/Page404/Page404.tsx
Comment (chelovek-zeml) [reply mw-webapp/src/pages/Page404/Page404.tsx:3]: It's useful because this text is a magic value. All magic values should be move out in consts.

File: mw-webapp/src/pages/Page404/Page404.tsx
Comment (chelovek-zeml) [reply mw-webapp/src/pages/Page404/Page404.tsx:5]: can be named constant: TEXT_ERROR?

File: mw-webapp/src/pages/Page404/Page404.tsx
Comment (Howl404) [reply mw-webapp/src/pages/Page404/Page404.tsx]: "A "magic value" is a string or number used in a program that is essential to its proper function but provides no context or explanation for why it is what it is."

Only definition that i found, and by it - you cant really call this string a magic value, context is sufficient, will be happy to see other definition

File: mw-webapp/src/pages/Page404/Page404.module.scss
Comment (tritonJS826) [reply mw-webapp/src/pages/Page404/Page404.module.scss:6]: page should have dynamic height depends on height of content

File: mw-webapp/src/pages/Page404/Page404.tsx
Comment (tritonJS826) [reply mw-webapp/src/pages/Page404/Page404.tsx:5]: 1. On my own opinion numbers at the end looks ok
2. could you add a link to main page to get ability to move on for users)

File: mw-webapp/src/pages/Page404/Page404.tsx
Comment (tritonJS826) [inline mw-webapp/src/pages/Page404/Page404.tsx:7]: I propose to use as different naming if possible. For example almost each component could contain container and it will be a little hard to find appropriate one when you are trying to debug etc. Much more easier to find  something more specific like errorPageContainer. It is just matter of code style, let's try this one

File: mw-webapp/src/styles/_variables.scss
Comment (tritonJS826) [reply mw-webapp/src/styles/_variables.scss:22]: 1. is it possible to use cameCase for variables?
2. please put comment before lines
3. Do we really need 5 sizes? get rid of unused

File: mw-webapp/src/pages/Page404/Page404.tsx
Comment (tritonJS826) [reply mw-webapp/src/pages/Page404/Page404.tsx]: @Howl404 agreed with you, but it is just a convention for the current project to move all these kinds of values to variables. In my opinion, it is simpler to put all variables in constants than teach everybody to differentiate without debates about what will be used somewhere else and what is the real magic number. BTW ,  you provide good 3 points above, I like them :) You could provide clear convention for what should be placed in separate variable and what should not and we can discuss  it later

File: mw-webapp/src/styles/_variables.scss
Comment (tritonJS826) [reply mw-webapp/src/styles/_variables.scss]: Should linter fix it automatically? Please check configuration

File: mw-webapp/src/pages/Page404/Page404.tsx
Comment (tritonJS826) [reply mw-webapp/src/pages/Page404/Page404.tsx:5]: dont see changes related to point 2

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

**Comment by tritonJS826** [comment]
I cant approve until you merge conflicts

**Comment by tritonJS826** [comment]
Don't see any changes related to the last review

File: mw-webapp/src/pages/Page404/Page404.tsx
Comment (tritonJS826) [reply mw-webapp/src/pages/Page404/Page404.tsx:5]: Still no changes

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

**Comment by chelovek-zeml** [comment]
I wrote above that we have a common header and there is a link to the main page there.

File: mw-webapp/src/pages/Page404/Page404.tsx
Comment (tritonJS826) [inline mw-webapp/src/pages/Page404/Page404.tsx:7]: Semantically each page should contain a heading, please add it before merging

File: mw-webapp/src/styles/_variables.scss
Comment (tritonJS826) [inline mw-webapp/src/styles/_variables.scss:18]: What about font size for title? Use mixin for title before merge

**Review by tritonJS826 — APPROVED** [review]

**Comment by chelovek-zeml** [comment]
Done

File: mw-webapp/src/styles/_variables.scss
Comment (Howl404) [reply mw-webapp/src/styles/_variables.scss]: 1

File: mw-webapp/src/pages/Page404/Page404.tsx
Comment (Ekaterina1994) [inline mw-webapp/src/pages/Page404/Page404.tsx:3]: We have component title for heading/ No need create new component

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

**Review by Ekaterina1994 — APPROVED** [review]

**Pull request merged**
